api/v1alpha1/bundleinstance_types.go

```diff
@@ -57,6 +57,8 @@ type BundleTemplate struct {
 	// Standard object's metadata.
 	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
 	// +optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Schemaless
```

Comment:
How come we don't want to validate the schema for this object? Is it to make the YAML UX more ergonomic, or to allow the client to pass in additional fields we don't necessarily reconcile on?

Reply:
I don't see anything like this in the PodTemplateSpec, so I wonder if/how they solved it for deployments/statefulsets/etc. Seems a little scary to allow arbitrary fields to be added and for us not to do validation of anything inside, unless I'm misinterpreting.

Reply:
> Seems a little scary to allow arbitrary fields to be added and for us not to do validation of anything inside, unless I'm misinterpreting

This seems valid but I'll caution that adding arbitrary metadata fields doesn't result in the plain provisioner respecting them. The current implementation only respects propagating the labels/annotations right now.

Reply:
Right, I guess my concern is the UX of "we'll let you create stuff in the BI template, but then we'll silently ignore some of it"

It'd be best, IMO, to block (via CRD schema or a webhook) BI creation/update if it contains something that we know we'll ignore or will cause a Bundle creation failure.

Reply:
Resolved in [slack](https://kubernetes.slack.com/archives/C038B7MF75M/p1651866914692939?thread_ts=1651865835.897279&cid=C038B7MF75M). Joe had discovered that there's a controller-gen marker for this, and we removed the kubebuilder preserveUnknownFields and schemaLess markers from these set of changes.